bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVHttp.java

```diff
@@ -0,0 +1,164 @@
+     *            request. Null if no parameters.
+     * @param soapAction Header attribute for SOAP ACTION: xxx
+     * @param connection Header attribut for CONNECTION: xxx
+     * @return String contents of the response for the POST request.
+     * @throws MagentaTVException
+     */
+    public String httpPOST(String host, String url, String postData, String soapAction, String connection)
+            throws MagentaTVException {
+        String httpResponse = "";
+        try {
+            Properties httpHeader = new Properties();
+            httpHeader.setProperty(HEADER_CONTENT_TYPE, CONTENT_TYPE_XML);
+            httpHeader.setProperty(HEADER_ACCEPT, "");
+            httpHeader.setProperty(HEADER_USER_AGENT, USER_AGENT);
+            httpHeader.setProperty(HEADER_HOST, host);
+            if (!soapAction.isEmpty()) {
+                httpHeader.setProperty(HEADER_SOAPACTION, soapAction);
+            }
+            if (!connection.isEmpty()) {
+                httpHeader.setProperty(HEADER_CONNECTION, connection);
+            }
+
+            logger.trace("POST {} - SoapAction={}, Data = {}", url, postData, soapAction);
+            InputStream dataStream = new ByteArrayInputStream(
+                    postData.getBytes(Charset.forName(StandardCharsets.UTF_8.name())));
+            httpResponse = HttpUtil.executeUrl(HttpMethod.POST, url, httpHeader, dataStream, null, NETWORK_TIMEOUT_MS);
+            logger.trace("POST {} - Response = {}", url, httpResponse);
+            return httpResponse;
+        } catch (IOException e) {
+            throw new MagentaTVException(e, "HTTP POST {0} failed, response={1}", url, httpResponse);
+        }
+    }
+
+    /**
+     * Send raw TCP data (SUBSCRIBE command)
+     *
+     * @param remoteIp receiver's IP
+     * @param remotePort destination port
+     * @param data data to send
+     * @return received response
+     * @throws IOException
+     */
+    public String sendData(String remoteIp, String remotePort, String data) throws MagentaTVException {
+        String response = "";
+        Socket socket = null;
+        String errorMessage = "";
+
+        try {
+            // logger.trace("Sending data to {}:{}: {}", remoteIp, remotePort, data);
+            socket = new Socket();
+            socket.setSoTimeout(4 * 1000); // set read timeout < 5s
+            socket.connect(new InetSocketAddress(remoteIp, Integer.parseInt(remotePort)), 3000);
+
+            OutputStream out = socket.getOutputStream();
+            PrintStream ps = new PrintStream(out, true);
+            ps.println(data);
+
+            InputStream in = socket.getInputStream();
+            BufferedReader buff = new BufferedReader(new InputStreamReader(in));
+
+            // wait until somthing to read is available or socket I/O fails (IOException)
+            int retry = NETWORK_TIMEOUT_MS / 50;
+            while (!buff.ready() && retry-- > 0) {
+                Thread.sleep(50);
+            }
```

Comment:
Don't do this, please rely on the timeout configured for the socket, it will throw an exception for you if a read takes too long so you don't need to wait for the stream to become ready.

Reply:
need to check that, as far as I remember there was a reason for that

Reply:
changed